libcache/cache.c

```diff
@@ -404,7 +405,6 @@ static ssize_t cache_fetchLine(cachectx_t *cache, cacheline_t *linePtr, const ui
 	while (left > 0) {
 		readCount = cache->ops.readCb(tempAddr, (unsigned char *)linePtr->data + position, left, cache->ops.ctx);
 		if (readCount <= 0) {
-			free(linePtr->data);
```

Comment:
I'm not convinced that we shouldn't free memory here. We got either EOF or some IO error on the device, chances seem slim to me that we'll be able to this line later